android/src/main/java/com/reactnativecommunity/webview/RNCWebView.java

```diff
@@ -184,7 +188,7 @@ public void setMessagingEnabled(boolean enabled) {
         messagingEnabled = enabled;
 
         if (enabled) {
-            addJavascriptInterface(createRNCWebViewBridge(this), JAVASCRIPT_INTERFACE);
+            addJavascriptInterface(getRNCWebViewBridge(this), JAVASCRIPT_INTERFACE);
         } else {
             removeJavascriptInterface(JAVASCRIPT_INTERFACE);
         }
```

Comment:
With this updated implementation, I'm concerned about the possibility of RNCWebViewBridge being removed from the webview if messaging is disabled, and I'm interested in suggestions for a preferred way to address them.

Reply:
Ah, yes. That is a valid concern I had not considered.

I could imagine perhaps refactoring this to have a separate `javascriptInterfaceEnabled` (or `webViewBridgeEnabled`) type of member variable that's used to manage the `addJavascriptInterface`/`removeJavascriptInterface` calls, then that would be set according to the combination of the `messagingEnabled`/`injectedObject` variables....?

(and the actual implementations would need to be guarded by their respective properties... e.g., `onMessage` shouldn't do anything if called when `messagingEnabled` is `false`...)

I'm still open to considering separate interfaces, I just thought the single interface would be cleaner.

Reply:
@TheAlmightyBob @jamonholmgren and I worked together to come up with a solution to this dilemma. Now, rather than removing the javascript interface, we check if messaging is enabled when a message comes through. There is a very narrow window where this might break existing code if a website were relying on checks for the existence of `window.RNCWebViewBridge` to determine whether or not it's expecting the app to catch messages, but in all other scenarios we could think of it shouldn't cause any issues.

Reply:
> There is a very narrow window where this might break existing code if a website were relying on checks for the existence of window.RNCWebViewBridge to determine whether or not it's expecting the app to catch messages

That does seem like a potentially breaking change, which I was trying to avoid with the idea of "still remove the interface if _neither_ `messagingEnabled` _nor_ `injectedObject` are set" (so the behavior would be unchanged if people don't touch `injectedObject`).

There's also the small possibility of this being a security concern... Google themselves [warn](https://developer.android.com/develop/ui/views/layout/webapps/webview#BindingJavaScript) that `addJavascriptInterface`
> can be a very useful feature or a dangerous security issue

(And I previously had to review `react-native-webview`'s use of it with my company's security team)

I don't think what you're proposing changes the _actual_ security risk, I just wonder if people would be concerned about `addJavascriptInterface` being invoked even if they're not enabling features that require it...


If @jamonholmgren / @Titozzz think that's all fine and are comfortable merging this as a breaking change, I won't object. But I'm not personally able to merge breaking changes.

Reply:
@TheAlmightyBob The way the new code works, if neither `messagingEnabled` (aka `onMessage`) nor `injectedObject` are set, the interface does not get added to the webview at all, so in that regard the behavior remains unchanged. It's only if `onMessage` is set and then subsequently unset where the behavior is different (an extreme edge case, IMO), other than using `injectedObject` obviously.

Reply:
Oh. Okay, I'll take a look at the new code later, my comment above was only responding to my interpretation of your comment, not the actual changes.

Reply:
Yeah it's a very edge-case situation where behavior would change (adding and then removing an existing `onMessage` prop, PLUS relying on the existence of `window.RNCWebVewBridge` to tell them ... something?). If a single user reports an issue I'd be extremely surprised. :joy:

We also protect against messages being sent in the code by checking `messagingEnabled` which is helpful for security concerns.